dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -332,18 +383,25 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                 }
             }
 
-            if place.is_indirect() && !context.is_mutating_use() {
+            if place.is_indirect()
+                && !context.is_mutating_use()
+                // reborrows do not apply, i.e. _2 = &(*_1)
+                && !matches!(self.assignment, Some((_, Rvalue::Ref(..))))
```

Comment:
I think this match and the following match should each get split into mini functions with a descriptive name or use bool variables with names, rather than adding comments in the middle of the conditional.

Reply:
this will be addressed in #552